core/src/elements/atomic/heading.rs

```diff
@@ -125,14 +114,20 @@ pub struct Heading {
     pub line_nr: usize,
 }
 
-pub enum Token<'a> {
+/// HeadingToken for the [`ElementParser`]
+pub enum HeadingToken<'a> {
```

Comment:
Should we use specific names for there? I think `Token` is also fine and we should simply not re-export everything from element's module.

Reply:
In PR #89 I just re-exported directly the types we used and it wasn't a problem. We can merge it, or leave it, up to you 👍🏻 

Resolve this convo when you've read it and do as you see fit 🤝

Reply:
Clippy did also not fail locally, but only in the GitHub action, so really strange.
But since both tokens have the same name, the warning made sense.

For now, I would leave it as is in this PR, but you are free to change it back to `Token` in your PR, and not re-export everything from elements.

I prefer your approach with `Token` and no re-exports, but would leave it up to you in PR #89